### PR TITLE
docs: Document new K8s controllers

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: November 15

Fixes redpanda-data/documentation-private#2018

Previews:

- [Decommission brokers](https://deploy-preview-107--redpanda-docs-preview.netlify.app/current/manage/kubernetes/decommission-brokers/#decommission-a-broker)
- [Install the Nodewatcher controller](https://deploy-preview-107--redpanda-docs-preview.netlify.app/current/manage/kubernetes/scale/#node-pvc)

### CODE REVIEW

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 1]
```diff
@@ -5,12 +5,22 @@
 
 When you decommission a broker, its partition replicas are reallocated across the remaining brokers and it is removed from the cluster. You may want to decommission a broker in the following circumstances:
 
+* You are removing a broker to decrease the size of the cluster, also known as scaling in.
```
Comment (alejandroEsc) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
* You are removing a broker to decrease the size of the cluster, also known as scaling down.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 2]
```diff
@@ -189,55 +199,286 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+----
+apiVersion: cluster.redpanda.com/v1alpha1
+kind: Redpanda
+metadata:
+  name: redpanda
+spec:
+  chartRef: {}
+  clusterSpec:
+    statefulset:
+      replicas: 5
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc:398]:
```suggestion
      replicas: 6
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 3]
```diff
@@ -126,43 +130,49 @@ Throughput is the primary measurement required to calculate future data storage
...
+In the example cluster, only six hours of data locally must be retained. Any older data can be moved to xref:manage:kubernetes/storage/tiered-storage.adoc[Tiered Storage] with a retention of one year. So each broker should have 1.2TB of storage available, taking into account both throughput and current data.
 
-Cost and use case requirements determine how much to spend on local disk capacity. Tiered Storage can help to both decrease costs and expand data retention capabilities. For details, see xref:manage:kubernetes/storage/tiered-storage.adoc[Tiered Storage].
+Cost and use case requirements determine how much to spend on local disk capacity. Tiered Storage can help to both decrease costs and expand data retention capabilities.
 
 [NOTE]
 ====
-At this point in the example, it remains unclear that we can scale down to five brokers. Current calculations are based on five brokers. Other broker counts can be used later as needed.
+At this point in the example, it remains unclear whether it is safe to scale in to five brokers. Current calculations are based on five brokers.
```
Comment (alejandroEsc) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
At this point in the example, it remains unclear whether it is safe to scale down to five brokers. Current calculations are based on five brokers.
```

again a suggestion

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 4]
```diff
@@ -189,55 +199,292 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+When you reduce the number of replicas, the controller terminates the Pod with the highest ordinal, removes its PVC, and then attempts to set the reclaim policy of the PV to `Retain`. Finally, the controller waits for the cluster state to become healthy before committing to decommissioning the broker that was running in the terminated Pod.
+
+NOTE: Always decommission one broker at a time.
+
+. Install the Decommission controller:
++
+[tabs]
+======
+Helm + Operator::
++
```
Comment (alejandroEsc) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc:257]: we should add that clusterscoped rbac rules are required.

**Review by alejandroEsc — COMMENTED** [review]
I have not finished reviewing as I keep getting randomized. The one thing i have seen from testing yesterday is that pods are rolling that i was not expecting to when scaling down. Going to check up on this.

**Comment by JakeSCahill** [comment]
> I have not finished reviewing as I keep getting randomized. The one thing i have seen from testing yesterday is that pods are rolling that i was not expecting to when scaling down. Going to check up on this.

Would this be because the brokers need to update their `seed_servers` list?

File: modules/manage/pages/kubernetes/decommission-brokers.adoc (hunk 4 above)
Comment (JakeSCahill) [reply modules/manage/pages/kubernetes/decommission-brokers.adoc:257]: is this not covered here?

- `rbac.createAdditionalControllerCRs=true`: Creates the required RBAC rules for the Redpanda Operator to monitor the StatefulSet and update PVCs and PVs.

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 5]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+  --namespace <namespace> \
+  --set image.repository=docker.redpanda.com/redpandadata/redpanda-operator \
+  --set image.tag=v{full-version} \
+  --create-namespace \
+  --set additionalCmdFlags={--additional-controllers="decommission"} \
+  --set rbac.createAdditionalControllerCRs=true
+----
++
+- `--additional-controllers="decommission"`: Enables the Decommission controller.
+- `rbac.createAdditionalControllerCRs=true`: Creates the requires RBAC rules for the Redpanda Operator to monitor the StatefulSet and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
- `rbac.createAdditionalControllerCRs=true`: Creates the required RBAC rules for the Redpanda Operator to monitor the StatefulSet and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 6]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+        run:
+          - "decommission"
+    rbac:
+      enabled: true
+----
+
+- `statefulset.replicas`: This example starts with a seven-broker Redpanda cluster.
+- `statefulset.sideCars.controllers.enabled`: Enables the controllers sidecar.
+- `statefulset.sideCars.controllers.run`: Enables the Decommission controller.
+- `rbac.enabled`: Creates the requires RBAC rules for the controller to monitor the StatefulSet and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
- `rbac.enabled`: Creates the required RBAC rules for the controller to monitor the StatefulSet and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 7]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+      run:
+        - "decommission"
+rbac:
+  enabled: true
+----
++
+- `statefulset.replicas`: This example starts with a seven-broker Redpanda cluster.
+- `statefulset.sideCars.controllers.enabled`: Enables the controllers sidecar.
+- `statefulset.sideCars.controllers.run`: Enables the Decommission controller.
+- `rbac.enabled`: Creates the requires RBAC rules for the controller to monitor the StatefulSet and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
- `rbac.enabled`: Creates the required RBAC rules for the controller to monitor the StatefulSet and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 8]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+  --set statefulset.replicas=7 \
+  --set statefulset.sideCars.controllers.enabled=true \
+  --set statefulset.sideCars.controllers.run={"decommission"} \
+  --set rbac.enabled=true
+----
++
+- `statefulset.replicas`: This example starts with a seven-broker Redpanda cluster.
+- `statefulset.sideCars.controllers.enabled`: Enables the controllers sidecar.
+- `statefulset.sideCars.controllers.run`: Enables the Decommission controller.
+- `rbac.enabled`: Creates the requires RBAC rules for the controller to monitor the StatefulSet and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
- `rbac.enabled`: Creates the required RBAC rules for the controller to monitor the StatefulSet and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/scale.adoc [hunk 9]
```diff
@@ -75,8 +98,183 @@ helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --crea
...
+  --namespace <namespace> \
+  --set image.repository=docker.redpanda.com/redpandadata/redpanda-operator \
+  --set image.tag=v{full-version} \
+  --create-namespace \
+  --set additionalCmdFlags={--additional-controllers="nodeWatcher"} \
+  --set rbac.createAdditionalControllerCRs=true
+----
++
+- `--additional-controllers="nodeWatcher"`: Enables the Nodewatcher controller.
+- `rbac.createAdditionalControllerCRs=true`: Creates the requires RBAC rules for the Redpanda Operator to monitor the Node resources and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/scale.adoc]:
```suggestion
- `rbac.createAdditionalControllerCRs=true`: Creates the required RBAC rules for the Redpanda Operator to monitor the Node resources and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/scale.adoc [hunk 10]
```diff
@@ -75,8 +98,183 @@ helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --crea
...
+          enabled: true
+        run:
+          - "nodeWatcher"
+    rbac:
+      enabled: true
+----
+
+- `statefulset.sideCars.controllers.enabled`: Enables the controllers sidecar.
+- `statefulset.sideCars.controllers.run`: Enables the Nodewatcher controller.
+- `rbac.enabled`: Creates the requires RBAC rules for the controller to monitor the Node resources and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/scale.adoc]:
```suggestion
- `rbac.enabled`: Creates the required RBAC rules for the controller to monitor the Node resources and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/scale.adoc [hunk 11]
```diff
@@ -75,8 +98,183 @@ helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --crea
...
+      enabled: true
+      run:
+        - "nodeWatcher"
+rbac:
+  enabled: true
+----
++
+- `statefulset.sideCars.controllers.enabled`: Enables the controllers sidecar.
+- `statefulset.sideCars.controllers.run`: Enables the Nodewatcher controller.
+- `rbac.enabled`: Creates the requires RBAC rules for the controller to monitor the Node resources and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/scale.adoc]:
```suggestion
- `rbac.enabled`: Creates the required RBAC rules for the controller to monitor the Node resources and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/scale.adoc [hunk 12]
```diff
@@ -75,8 +98,183 @@ helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --crea
...
+  --namespace <namespace> \
+  --create-namespace \
+  --set statefulset.sideCars.controllers.enabled=true \
+  --set statefulset.sideCars.controllers.run={"nodeWatcher"} \
+  --set rbac.enabled=true
+----
++
+- `statefulset.sideCars.controllers.enabled`: Enables the controllers sidecar.
+- `statefulset.sideCars.controllers.run`: Enables the Nodewatcher controller.
+- `rbac.enabled`: Creates the requires RBAC rules for the controller to monitor the Node resources and update PVCs and PVs.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/scale.adoc]:
```suggestion
- `rbac.enabled`: Creates the required RBAC rules for the controller to monitor the Node resources and update PVCs and PVs.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc (hunk 4 above)
Comment (alejandroEsc) [reply modules/manage/pages/kubernetes/decommission-brokers.adoc:257]: ok!

**Comment by alejandroEsc** [comment]
> > I have not finished reviewing as I keep getting randomized. The one thing i have seen from testing yesterday is that pods are rolling that i was not expecting to when scaling down. Going to check up on this.
> 
> Would this be because the brokers need to update their `seed_servers` list?

I thought this would be the case but doesnt seem to be. Something else is triggering the restarts.

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 13]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+----
++
+If you're running the Decommission controller as a sidecar:
++
+[,bash]
+----
+kubectl logs <pod-name> --namespace <namespace> -c redpanda-controllers
+----
+
+You can repeat this procedure to scale in to 5 brokers.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
You can repeat this procedure to scale down to 5 brokers.
```

File: modules/manage/pages/kubernetes/scale.adoc [hunk 14]
```diff
@@ -75,8 +98,183 @@ helm upgrade --install redpanda redpanda/redpanda --namespace <namespace> --crea
...
+=== Scale in
+
+Scaling in is the process of removing brokers from your Redpanda cluster. You may want to remove brokers for cost reduction and resource optimization.
+
+To scale in a Redpanda cluster, you must decommission the brokers that you want to remove before updating the `statefulset.replica` setting in the Helm values. See xref:manage:kubernetes/decommission-brokers.adoc[].
+
+[[node-pvc]]
+== Install the Nodewatcher controller
+
+The Nodewatcher controller manages the lifecycle of PersistentVolumes (PVs) and PersistentVolumeClaims (PVCs) for Redpanda clusters when worker nodes are deleted. When the controller detects that a Node resource is not available, it sets the reclaim policy of the PV to `Retain`, ensuring no data loss. Concurrently, it orchestrates the deletion of the PVC, which allows the Redpanda broker that was previously running on the deleted worker node to be rescheduled onto new, operational nodes.
```
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/scale.adoc]:
```suggestion
The Nodewatcher controller manages the lifecycle of PersistentVolumes (PVs) and PersistentVolumeClaims (PVCs) for Redpanda clusters. When the controller detects that a Node resource is not available, it sets the reclaim policy of the PV to `Retain`, ensuring no data loss. Concurrently, it orchestrates the deletion of the PVC, which allows the Redpanda broker that was previously running on the deleted worker node to be rescheduled onto new, operational nodes.
```

**Review by alejandroEsc — APPROVED** [review]
lgtm

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 15]
```diff
@@ -72,17 +76,17 @@ ID    HOST                          PORT   RACK
...
 
 === Cost
 
-Infrastructure costs increase with each broker, so adding a broker means an additional instance to pay for. In this example we deploy to GKE on https://gcloud-compute.com/n2-standard-8.html[n2-standard-8^] GCP instances. This means that the instance cost of the cluster is &dollar;1925 per month. Dropping down to 5 brokers would save &dollar;550 per month, and dropping down to 3 brokers would save &dollar;1100 per month. Of course, there are other costs to consider, but they won't be as impacted by changing the broker count.
+Infrastructure costs increase with each broker, so adding a broker means an additional instance to pay for. For example, if you deploy Redpanda on GKE on https://gcloud-compute.com/n2-standard-8.html[n2-standard-8^] GCP instances, the instance cost of the cluster is $1925 per month. Reducing the number of brokers to five would save $550 per month, and reducing it further to three brokers would save $1100 per month. You must also consider other costs, but they won't be as impacted by changing the broker count.
```
Comment (Feediver1) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
Infrastructure costs increase with each broker, so adding a broker means an additional instance cost. For example, if you deploy Redpanda on GKE on https://gcloud-compute.com/n2-standard-8.html[n2-standard-8^] GCP instances, the instance cost of the cluster is $1925 per month. Reducing the number of brokers to five would save $550 per month, and reducing it further to three brokers would save $1100 per month. You must also consider other costs, but they won't be as impacted by changing the broker count.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 16]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
 ```
 ====
 
-In this example each broker has 12 cores available. If you plan to scale down to five broker, then you would have 40 cores available, which means that your cluster is limited by core count to 40K partitions (well above the current 3018 partitions).
+In this example each broker has 8 cores available. If you plan to scale in to five broker, then you would have 40 cores available, which means that your cluster is limited by core count to 40K partitions, which exceeds the current 3018 partitions.
```
Comment (Feediver1) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
In this example, each broker has eight cores available. If you plan to scale down to five brokers, then you would have 40 cores available, which means that your cluster is limited by core count to 40K partitions, which exceeds the current 3018 partitions.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 17]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
 * At least four brokers are required based on availability.
-* Cost is not a limiting factor in this example, but lower cost (and lower broker count) is always best.
-* At least 1.2TB of data resides on each broker (if spread across five brokers). This falls within the 1.5TB of local storage available in the example.
+* Cost is not a limiting factor in this example, but lower cost and lower broker count is always best.
+* At least 1.2TB of data resides on each broker when spread across five brokers. This falls within the 1.5TB of local storage available in the example.
 * At least five brokers are required based on the highest replication factor across all topics.
 * At 3018 partitions, the partition count is so low as to not be a determining factor in broker count (a single broker in this example environment could handle many more partitions).
 
-So the primary limitation consideration is the replication factor of five, meaning that you could scale down to five brokers at minimum.
+So the primary limitation consideration is the replication factor of five, meaning that you could scale in to five brokers at minimum.
```
Comment (Feediver1) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]: I notice that here you write it out (five), but in other contexts above ("8 cores"), you use a numeral. Wondering why--I'd make it consistent. I believe the style guidance is any number less than 9 is written out, and any over 9 you just write the numeral.

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 18]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+metadata:
+  name: redpanda
+spec:
+  chartRef: {}
+  clusterSpec:
+    statefulset:
+      replicas: 7
+----
++
+- `statefulset.replicas`: This example starts with a seven-broker Redpanda cluster.
```
Comment (Feediver1) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc:301]:
```suggestion
- `statefulset.replicas`: This example starts with a 7-broker Redpanda cluster.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 19]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
+      sideCars:
+        controllers:
+          enabled: true
+        run:
+          - "decommission"
+    rbac:
+      enabled: true
+----
+
+- `statefulset.replicas`: This example starts with a seven-broker Redpanda cluster.
```
Comment (Feediver1) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc:326]:
```suggestion
- `statefulset.replicas`: This example starts with a 7-broker Redpanda cluster.
```

File: modules/manage/pages/kubernetes/decommission-brokers.adoc [hunk 20]
```diff
@@ -189,55 +199,294 @@ NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
...
-  rpk cluster info \
-    -X brokers=<broker-url>:<kafka-api-port> \
-    --tls-enabled \
-    --tls-truststore <path-to-kafka-api-ca-certificate>
+kubectl exec redpanda-0 --namespace <namespace> -- rpk cluster health
 ```
 
---
-TLS Disabled::
+. Decrease the number of replicas by one:
```
Comment (Feediver1) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc:386]:
```suggestion
. Decrease the number of replicas by 1:
```

**Review by Feediver1 — APPROVED** [review]
LGTM

File: modules/manage/pages/kubernetes/decommission-brokers.adoc (hunk 17 above)
Comment (JakeSCahill) [inline modules/manage/pages/kubernetes/decommission-brokers.adoc]:
```suggestion
So the primary limitation consideration is the replication factor of five, meaning that you could scale down to five brokers at minimum.
```

**Pull request merged**